src/main/java/slimeknights/tconstruct/smeltery/block/entity/CastingTankBlockEntity.java

```diff
@@ -0,0 +1,367 @@
+import net.minecraft.world.entity.player.Player;
+import net.minecraft.world.inventory.AbstractContainerMenu;
+import net.minecraft.world.item.BucketItem;
+import net.minecraft.world.item.Item;
+import net.minecraft.world.item.ItemStack;
+import net.minecraft.world.level.Level;
+import net.minecraft.world.level.block.Block;
+import net.minecraft.world.level.block.entity.BlockEntityType;
+import net.minecraft.world.level.block.state.BlockState;
+import net.minecraft.world.phys.BlockHitResult;
+import net.minecraftforge.client.model.data.ModelData;
+import net.minecraftforge.common.capabilities.Capability;
+import net.minecraftforge.common.capabilities.ForgeCapabilities;
+import net.minecraftforge.common.util.LazyOptional;
+import net.minecraftforge.fluids.FluidStack;
+import net.minecraftforge.fluids.FluidType;
+import net.minecraftforge.fluids.capability.IFluidHandler;
+import net.minecraftforge.fluids.capability.IFluidHandlerItem;
+import net.minecraftforge.items.ItemHandlerHelper;
+import net.minecraftforge.items.wrapper.SidedInvWrapper;
+import slimeknights.mantle.fluid.FluidTransferHelper;
+import slimeknights.mantle.fluid.transfer.FluidContainerTransferManager;
+import slimeknights.mantle.fluid.transfer.IFluidContainerTransfer;
+import slimeknights.tconstruct.TConstruct;
+import slimeknights.tconstruct.library.client.model.ModelProperties;
+import slimeknights.tconstruct.library.fluid.FluidTankAnimated;
+import slimeknights.tconstruct.library.utils.NBTTags;
+import slimeknights.tconstruct.shared.block.entity.TableBlockEntity;
+import slimeknights.tconstruct.smeltery.TinkerSmeltery;
+import slimeknights.tconstruct.smeltery.item.TankItem;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+
+public class CastingTankBlockEntity extends TableBlockEntity implements ITankBlockEntity, WorldlyContainer {
+  /** Max capacity for the tank */
+  public static final int DEFAULT_CAPACITY = FluidType.BUCKET_VOLUME * 4;
+  // slots
+  public static final int INPUT = 0;
+  public static final int OUTPUT = 1;
+  private static final Component NAME = TConstruct.makeTranslation("gui", "casting");
+
+  /** Internal fluid tank instance */
+  @Getter
+  protected final FluidTankAnimated tank;
+  /** Capability holder for the tank */
+  private final LazyOptional<IFluidHandler> fluidHolder;
+  /** Last comparator strength to reduce block updates */
+  @Getter @Setter
+  private int lastStrength = -1;
+
+  /**
+   * Gets the capacity for the given block
+   * @param block  block
+   * @return  Capacity
+   */
+  public static int getCapacity(Block block) {
+    return DEFAULT_CAPACITY;
+  }
+
+  /**
+   * Gets the capacity for the given item
+   * @param item  item
+   * @return  Capacity
+   */
+  public static int getCapacity(Item item) {
+    return DEFAULT_CAPACITY;
+  }
+
+  public CastingTankBlockEntity(BlockPos pos, BlockState state) {
+    this(pos, state, (ITankBlock) state.getBlock());
```

Comment:
No need for an unchecked cast, just do what I did this in `TankBlockEntity`.

Sure, it shouldn't happen, but I've seen people do hacky stuff with block entities giving them the wrong block.